pkgs/development/python-modules/jenkinsapi/default.nix

```
@@ -19,14 +24,19 @@ buildPythonPackage rec {
   };
 
   propagatedBuildInputs = [ pytz requests ];
-  buildInputs = [ coverage mock nose unittest2 ];
+  checkInputs = [ mock nose unittest2 pycodestyle pylint pytest pytest-mock requests-kerberos toml ];
```

looks like they use pytest as the test runner
```suggestion
  checkInputs = [ mock pycodestyle pylint pytest pytest-mock requests-kerberos toml ];
```
unittest2 may still be needed for python2

---

mmmaybe. I can't actually built it for python2, and it seems like I might have stumbled onto a deeper issue:
```
nix-env --show-trace -f default.nix -iA python2Packages.jenkinsapi
installing 'python2.7-jenkinsapi-0.3.11'
error: while evaluating the attribute 'drvPath' at /home/rski/Code/nixpkgs/lib/customisation.nix:163:7:
while evaluating the attribute 'nativeBuildInputs' of the derivation 'python2.7-jenkinsapi-0.3.11' at /home/rski/Code/nixpkgs/pkgs/development/interpreters/python/mk-python-derivation.nix:108:3:
while evaluating the attribute 'out.outPath' at /home/rski/Code/nixpkgs/lib/customisation.nix:156:13:
while evaluating the attribute 'nativeBuildInputs' of the derivation 'python2.7-pylint-1.9.5' at /home/rski/Code/nixpkgs/pkgs/development/interpreters/python/mk-python-derivation.nix:108:3:
while evaluating 'getOutput' at /home/rski/Code/nixpkgs/lib/attrsets.nix:464:23, called from undefined position:
while evaluating 'callPackageWith' at /home/rski/Code/nixpkgs/lib/customisation.nix:117:35, called from /home/rski/Code/nixpkgs/pkgs/top-level/python-packages.nix:5318:15:
while evaluating 'makeOverridable' at /home/rski/Code/nixpkgs/lib/customisation.nix:67:24, called from /home/rski/Code/nixpkgs/lib/customisation.nix:121:8:
while evaluating anonymous function at /home/rski/Code/nixpkgs/pkgs/development/python-modules/pyenchant/default.nix:1:1, called from /home/rski/Code/nixpkgs/lib/customisation.nix:69:16:
while evaluating 'makeOverridablePythonPackage' at /home/rski/Code/nixpkgs/pkgs/top-level/python-packages.nix:31:37, called from /home/rski/Code/nixpkgs/pkgs/development/python-modules/pyenchant/default.nix:8:1:
while evaluating 'makeOverridable' at /home/rski/Code/nixpkgs/lib/customisation.nix:67:24, called from /home/rski/Code/nixpkgs/pkgs/top-level/python-packages.nix:33:12:
while evaluating anonymous function at /home/rski/Code/nixpkgs/pkgs/development/interpreters/python/mk-python-derivation.nix:31:1, called from /home/rski/Code/nixpkgs/lib/customisation.nix:69:16:
pyenchant-3.1.1 not supported for interpreter python2.7
```
```
nix-env --show-trace -f default.nix -iA python27Packages.pylint
installing 'python2.7-pylint-1.9.5'
error: while evaluating the attribute 'drvPath' at /home/rski/Code/nixpkgs/lib/customisation.nix:163:7:
while evaluating the attribute 'nativeBuildInputs' of the derivation 'python2.7-pylint-1.9.5' at /home/rski/Code/nixpkgs/pkgs/development/interpreters/python/mk-python-derivation.nix:108:3:
while evaluating 'getOutput' at /home/rski/Code/nixpkgs/lib/attrsets.nix:464:23, called from undefined position:
while evaluating 'callPackageWith' at /home/rski/Code/nixpkgs/lib/customisation.nix:117:35, called from /home/rski/Code/nixpkgs/pkgs/top-level/python-packages.nix:5318:15:
while evaluating 'makeOverridable' at /home/rski/Code/nixpkgs/lib/customisation.nix:67:24, called from /home/rski/Code/nixpkgs/lib/customisation.nix:121:8:
while evaluating anonymous function at /home/rski/Code/nixpkgs/pkgs/development/python-modules/pyenchant/default.nix:1:1, called from /home/rski/Code/nixpkgs/lib/customisation.nix:69:16:
while evaluating 'makeOverridablePythonPackage' at /home/rski/Code/nixpkgs/pkgs/top-level/python-packages.nix:31:37, called from /home/rski/Code/nixpkgs/pkgs/development/python-modules/pyenchant/default.nix:8:1:
while evaluating 'makeOverridable' at /home/rski/Code/nixpkgs/lib/customisation.nix:67:24, called from /home/rski/Code/nixpkgs/pkgs/top-level/python-packages.nix:33:12:
while evaluating anonymous function at /home/rski/Code/nixpkgs/pkgs/development/interpreters/python/mk-python-derivation.nix:31:1, called from /home/rski/Code/nixpkgs/lib/customisation.nix:69:16:
pyenchant-3.1.1 not supported for interpreter python2.7
```
so it's not something i have done.
pyenchant has:
```
  disabled = isPy27;
```
and the !python2 pylint doesn't depend on it at all, https://github.com/NixOS/nixpkgs/commit/eb0d5fc3f076ad8fbac43158e77277066bbe28b9. I think I need to open a pr to also remove it from the 1.9 inputs.

---

yup,
```
nix-env -iA unstable.python27Packages.pylint
replacing old 'python2.7-pylint-1.9.5'
installing 'python2.7-pylint-1.9.5'
error: pyenchant-3.1.1 not supported for interpreter python2.7
(use '--show-trace' to show detailed location information)
```
https://github.com/NixOS/nixpkgs/pull/91721

---

ok, I updated the PR and removed the needless inputs. I cherry picked the pylint change to test it on this branch and python2 works, but I'll keep them separate PRs